### ISSUE TITLE
`trailing_closure` conflicts with `multiple_closures_with_trailing_closure`

### ISSUE THREAD

**Issue by hongping-quora** [realm/SwiftLint#5988]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

```swift
func myFunc1(
    a1 _: Int,
    a2 _: (Int) -> Void,
    a3 _: String,
    a4 _: (String) -> Void
) {}

// This function triggers a violation:
// warning: Trailing Closure Violation: Trailing closure syntax should be used whenever possible (trailing_closure)
func myFunc2() {
    myFunc1(
        a1: 1,
        a2: { _ in },
        a3: "2",
        a4: { _ in }
    )
}

// If I try using trailing closure, I get:
// Multiple Closures with Trailing Closure Violation: Trailing closure syntax should not be used when passing more than one closure argument (multiple_closures_with_trailing_closure)
func myFunc3() {
    myFunc1(
        a1: 1,
        a2: { _ in },
        a3: "2"
    ) {
        _ in
    }
}


```

Mention the command or other SwiftLint integration method that caused the issue. Include stack traces or command output.

```bash
$ swiftlint 
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)
  * 0.58.2
* Xcode version (run `xcodebuild -version` to be sure)
  * Xcode 16.2, Build version 16C5032a
* Installation method used (Homebrew, CocoaPods, building from source, etc)
  * brew install swiftlint
* Configuration file:

```yml
opt_in_rules:
  - trailing_closure
```

Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? If so, paste their
relative paths and respective contents.
No

**Comment by SimplyDanny** [comment]
The two rules contradicting each other doesn't mean any of them works incorrectly. They do what they're intended for, whether that makes sense or not. In general, it's not a goal that all rules work well together and do not cause conflicts. This is one example combination.

Best is to disable one of the rules locally if you otherwise prefer to leave it enabled.

**Comment by hongping-quora** [comment]
In the documentation:
https://realm.github.io/SwiftLint/trailing_closure.html

It seems having multiple closures and not using trailing closure shouldn't trigger the violation?

Under: Non Triggering Examples
```
foo(param1: { _ in false }, param2: { _ in true }, param3: { _ in false })
```

This is very similar to what `myFunc2` is doing?

```
func myFunc2() {
    myFunc1(
        a1: 1,
        a2: { _ in },
        a3: "2",
        a4: { _ in }
    )
}

```

**Comment by hongping-quora** [comment]
Thanks for the explanation!

I just realized `myFunc2` matches this in the triggering example in the documentation.
https://realm.github.io/SwiftLint/trailing_closure.html

```
foo.something(param1: { _ in true }, param2: 0, param3: ↓{ _ in false })
```

So one possible fix might be to reorder the parameters so that the closures arguments are towards the end.

```

func myFunc4(
    a1 _: Int,
    a2 _: String,
    a3 _: (Int) -> Void,
    a4 _: (String) -> Void
) {}

func myFunc5() {
    myFunc4(
        a1: 1,
        a2: "2",
        a3: { _ in },
        a4: { _ in }
    )
}
```

**Comment by SimplyDanny** [comment]
> So one possible fix might be to reorder the parameters so that the closures arguments are towards the end.

Yes, that would silence `trailing_closure`. But don't feel forced to change your code only to satisfy a linter complaint if it doesn't make any sense in a specific setup. Most rules are opinionated and shouldn't be perceived as a given law that must be followed.

**Comment by hongping-quora** [comment]
Thanks for looking into this and the response. Feel free to close the task if there isn't any further action required. I can close this too, just wanted to check if it is ok for the reporter to do so, or do you typically do that on your end? Thanks!

**Comment by SimplyDanny** [comment]
> Thanks for looking into this and the response. Feel free to close the task if there isn't any further action required. I can close this too, just wanted to check if it is ok for the reporter to do so, or do you typically do that on your end? Thanks!

It is okay for the reporter to close the issue if the problem they had got resolved. Especially the ones labeled with "help" typically fall into this category.
